src/Microsoft.AspNet.Security/Infrastructure/SecurityHelper.cs

```diff
@@ -38,26 +40,26 @@ public static void AddUserIdentity([NotNull] HttpContext context, [NotNull] IIde
             context.User = newClaimsPrincipal;
         }
 
-        public static bool LookupChallenge(IEnumerable<string> authenticationTypes, string authenticationType, AuthenticationMode authenticationMode)
+        public static bool LookupChallenge(IEnumerable<string> authenticationSchemes, string authenticationScheme, AuthenticationMode authenticationMode)
         {
-            bool challengeHasAuthenticationTypes = authenticationTypes != null && authenticationTypes.Any();
-            if (!challengeHasAuthenticationTypes)
+            bool challengeHasAuthenticationSchemes = authenticationSchemes != null && authenticationSchemes.Any();
+            if (!challengeHasAuthenticationSchemes)
             {
                 return authenticationMode == AuthenticationMode.Active;
             }
-            return authenticationTypes.Contains(authenticationType, StringComparer.Ordinal);
+            return authenticationSchemes.Contains(authenticationScheme, StringComparer.Ordinal);
         }
 
         /// <summary>
         /// Find response sign-in details for a specific authentication middleware
         /// </summary>
-        /// <param name="authenticationType">The authentication type to look for</param>
-        public static bool LookupSignIn(IEnumerable<ClaimsIdentity> identities, string authenticationType, out ClaimsIdentity identity)
+        /// <param name="authenticationScheme">The authentication type to look for</param>
+        public static bool LookupSignIn(IEnumerable<ClaimsIdentity> identities, string authenticationScheme, out ClaimsIdentity identity)
         {
             identity = null;
             foreach (var claimsIdentity in identities)
             {
-                if (string.Equals(authenticationType, claimsIdentity.AuthenticationType, StringComparison.Ordinal))
+                if (string.Equals(authenticationScheme, claimsIdentity.AuthenticationType, StringComparison.Ordinal))
                 {
                     identity = claimsIdentity;
```

Comment:
OK, but it is related to how the Authenticated CP is put in the session. LookupSignIn sets the outparam to a single identity. My research indicates that results in a single identity being persisted.  Is this correct?